### PR TITLE
Restore ownership contracts

### PR DESCRIPTION
**Pull request by ajsantander**
**TL;DR**
A few days ago, we removed Ownable.sol and OwnableStorage.sol because they weren't being used by any of the other contracts within core directly. However, these contracts are needed externally by projects that use core, and the classic OpenZeppelin Ownable cannot replace it. 

**Explanation**
OZ's Ownable can only transfer ownership via `transferOwnership(address newOwner) public onlyOwner`. That is, it can only transfer ownership when there is an external call with `msg.sender == tx.origin`.

With core, ProjectController's `createAndCall()` might call an initialize method where the ownership is transferred from the deployer of an implementation to `msg.sender`. That is, `msg.sender != tx.origin`. At the time of the initializer execution, `msg.sender` is the controller.

To resolve this, the contracts being restored have the ability to transfer ownership via an `internal` `setOwner` call that has no requirements about the sender. This gives the contract itself the faculty to assign ownership, which is a feature missing from OZ's Ownable, and essential for contracts that need the proxy to assign ownership in an "on-chain constructor": the `initialize` function.

### CODE REVIEW

**Comment by spalladino** [comment]
-1 to duplicating the Ownable code. Given that the internal `owner` variable can be accessed from within any contract that extends Ownable, I'd go with just manually setting `owner = newOwner`, instead of calling `transferOwnership`. For reference, this is how we handled it [in the kernel registry](https://github.com/zeppelinos/kernel-registry/blob/master/contracts/ZepToken.sol#L17).

Also, restoring the Ownable contracts here causes conflicts if the target project also uses OZ, since solidity/truffle are not ok having several contracts with the same name.

**Comment by facuspagnuolo** [comment]
I agree with palla, can we close it then @ajsantander ?

**Comment by ajsantander** [comment]
Agreed!
